app/assets/js/app.ts

```diff
@@ -1,26 +1,5 @@
 import "~css/app.css";
-import "@docsearch/css";
+import defo from "@icelab/defo";
+import { views } from "./views";
 
-import docsearch from "@docsearch/js";
-
-interface AppSettings {
-  docsearch_app_id: string;
-  docsearch_api_key: string;
-  docsearch_index_name: string;
-}
-
-declare global {
-  interface Window {
-    app_settings: AppSettings;
-  }
-}
-
-docsearch({
-  container: "#docsearch",
-  appId: window.app_settings.docsearch_app_id,
-  apiKey: window.app_settings.docsearch_api_key,
-  indexName: window.app_settings.docsearch_index_name,
-  searchParameters: {
-    facetFilters: ["version:latest"],
-  },
-});
+defo({ views });
```

Comment:
I haven’t configured a `prefix` here, but we could if we wanted the bindings to be something other than `data-defo-*`.